docs/how-to/hip_runtime_api/asynchronous.rst

```diff
@@ -0,0 +1,488 @@
+Asynchronous concurrent execution
+*******************************************************************************
+
+Asynchronous concurrent execution is important for efficient parallelism and
+resource utilization, with techniques such as overlapping computation and data
+transfer, managing concurrent kernel execution with streams on single or
+multiple devices, or using HIP graphs.
+
+Streams and concurrent execution
+===============================================================================
+
+All asynchronous APIs, such as kernel execution, data movement and potentially
+data allocation/freeing all happen in the context of device streams.
+
+Streams are FIFO buffers of commands to execute in order on a given device.
+Commands which enqueue tasks on a stream all return promptly and the task is
+executed asynchronously. Multiple streams can point to the same device and
+those streams might be fed from multiple concurrent host-side threads. Execution
+on multiple streams might be concurrent but isn't required to be.
+
+Managing streams
+-------------------------------------------------------------------------------
+
+Streams enable the overlap of computation and data transfer, ensuring
+continuous GPU activity. 
+
+To create a stream, the following functions are used, each defining a handle
+to the newly created stream:
+
+- :cpp:func:`hipStreamCreate`: Creates a stream with default settings.
+- :cpp:func:`hipStreamCreateWithFlags`: Creates a stream, with specific
+  flags, listed below, enabling more control over stream behavior:
+
+  - ``hipStreamDefault``: creates a default stream suitable for most
+    operations. The default stream is a blocking operation.
+  - ``hipStreamNonBlocking``: creates a non-blocking stream, allowing
+    concurrent execution of operations. It ensures that tasks can run
+    simultaneously without waiting for each other to complete, thus improving
+    overall performance.
+
+- :cpp:func:`hipStreamCreateWithPriority`: Allows creating a stream with a
+  specified priority, enabling prioritization of certain tasks.
+
+The :cpp:func:`hipStreamSynchronize` function is used to block the calling host
+thread until all previously submitted tasks in a specified HIP stream have
+completed. It ensures that all operations in the given stream, such as kernel
+executions or memory transfers, are finished before the host thread proceeds.
+
+.. note::
+
+  If the :cpp:func:`hipStreamSynchronize` function input stream is 0 (or the
+  default stream), it waits for all operations in the default stream to
+  complete.
+
+Concurrent execution between host and device
+-------------------------------------------------------------------------------
+
+Concurrent execution between the host (CPU) and device (GPU) allows the CPU to
+perform other tasks while the GPU is executing kernels. Kernels can be launched
+asynchronously using ``hipLaunchKernelDefault`` with a stream, enabling the CPU
```

Comment:
What is hipLaunchKernelDefault? Where is it defined?

Reply:
Fixed this sentence.